### DISCUSSION TITLE
Separate server & client

### DISCUSSION THREAD

**Discussion by albingroen** [trpc/trpc#1363] [Q&A]
So, I think TRPC is pretty sick, and it works great for serverless setups like Next.js + API Routes, etc.

But, what if I want to use it with 1 server and 4 clients that live in separate codebases/code repositories.

How would I import the type definitions from the server into all of these clients? Is this not a case TRPC wants to support? It's pretty common in organizations in my experience.

**Comment by sachinraja** [comment]
Seems similar to #1362. You could look at the discussion there and https://github.com/KATT/zart for an example (1 server and 2 clients). You'd probably have to generate a type definition file here though (with `tsc`) and import the `AppRouter` type from there.

Although that would take away one of the main benefits of trpc (instant feedback, no generators needed). You could try exporting `AppRouter` and linking the server package using your package manager to the clients. Not sure if the language server would pick up new types on changes though.

**Reply by pixelpax** [reply]
What about linking a local package as a dev dependency?

**Comment by Sean-Xiao-Reckon** [comment]
in this case i think it's better to use GraphQL
